### PR TITLE
Changed behavior of `readJournalFiles` to be identical to `readJournalFile` for singleton lists

### PR DESCRIPTION
**Pull request by johannesgerer**
This solves  #436.

### CODE REVIEW

File: hledger-lib/Hledger/Read.hs [hunk 1]
```diff
@@ -126,8 +127,11 @@ defaultJournalPath = do
 --
 readJournalFiles :: Maybe StorageFormat -> Maybe FilePath -> Bool -> [PrefixedFilePath] -> IO (Either String Journal)
 readJournalFiles mformat mrulesfile assrt prefixedfiles = do
-  (either Left (Right . mconcat) . sequence)
+  (right mconcat1 . sequence)
     <$> mapM (readJournalFile mformat mrulesfile assrt) prefixedfiles
+  where mconcat1 :: Monoid t => [t] -> t
+        mconcat1 [] = mempty
+        mconcat1 x = foldr1 mappend x
```
Comment (simonmichael) [inline hledger-lib/Hledger/Read.hs]: I often hear foldr is the wrong fold to use.. is it the right one here, do you think ?

Comment (simonmichael) [inline hledger-lib/Hledger/Read.hs]: Is the haddock for readJournalFiles still correct (about the parse state) ?

Comment (johannesgerer) [inline hledger-lib/Hledger/Read.hs]: the old `mconcat` also used it, so no change here. At least it is explicitly visible now.

Comment (johannesgerer) [inline hledger-lib/Hledger/Read.hs]: Yes the docs are correct. Actually, only the new version of the function does what the docs say, as the parse state from the last file now lives on, due to the use of `mconcat1`, which only uses `mempty` for empty lists.

**Comment by johannesgerer** [comment]
This is marked as 'merged', yet i cannot find the commit in branch master. Why?

**Comment by simonmichael** [comment]
I don't know! I just clicked the merge button on github. I've redone it manually, thanks.

**Pull request merged**